packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/sass-language.ts

```diff
@@ -97,31 +103,49 @@ async function compileString(
             url,
             options: FileImporterWithRequestContextOptions,
           ): Promise<URL | null> => {
+            let resolved = resolutionCache.get(url);
+            if (resolved !== undefined) {
+              return resolved;
+            }
+
+            // Default to not found
+            resolved = null;
+
             const result = await resolveUrl(url, options);
             if (result.path) {
-              return pathToFileURL(result.path);
+              resolved = pathToFileURL(result.path);
+            } else {
+              // Check for package deep imports
+              const parts = url.split('/');
+              const hasScope = parts.length >= 2 && parts[0].startsWith('@');
+              const [nameOrScope, nameOrFirstPath, ...pathPart] = parts;
+              const packageName = hasScope ? `${nameOrScope}/${nameOrFirstPath}` : nameOrScope;
+
+              // Caching package root locations is particularly beneficial for `@material/*` packages
+              // which extensively use deep imports.
+              let packageRoot = packageRootCache.get(packageName);
+              if (packageRoot === undefined) {
+                const packageResult = await resolveUrl(packageName + '/package.json', options);
+                if (packageResult.path) {
+                  packageRoot = dirname(packageResult.path);
+                  packageRootCache.set(packageName, packageRoot);
+                }
+              }
+
+              if (packageRoot) {
```

Comment:
Question: Under what circumstance would we not have a `packageRoot` for an associated `packageName`? Would that be if the user typos a Sass import and specifies a non-existent package?

Reply:
Could be a typo in which case there will be an error generated at the end of the full resolution process.  It could also not be a package and is handled by a later resolution stage.

Reply:
Does Sass require `./` for relative imports or not (I'm guessing not)? If it does require `./`, we should probably check for that explicitly to limit this logic to real packages. If it doesn't then I get that we need this check and should probably include a comment mentioning that we're covering the non-package case.

Reply:
It does not unfortunately. A bare specifier could potentially be relative.